python/whylogs/core/validators/condition_validator.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from itertools import chain
+from typing import Any, Callable, Dict, List
+
+from whylogs.core.metrics.condition_count_metric import Condition
+from whylogs.core.preprocessing import PreprocessedColumn
+from whylogs.core.validators.validator import Validator
+
+
+@dataclass
+class ConditionValidator(Validator):
+    def __init__(
+        self, name: str, conditions: Dict[str, Condition], actions=List[Callable[[str, str, Any], None]]
+    ) -> None:
+        self.conditions = conditions
+        self.name = name
+        self.actions = actions
+        self.total: int = 0
+        self.failures: Dict[str, int] = dict()
```

Comment:
With dataclass you don't have to write most of that init method:
```suggestion
        name: str
        conditions: Dict[str, Condition]
        actions=List[Callable[[str, str, Any], None]]
        total: int = 0
        failures: Dict[str, int] = field(default_factory=dict)
```
Not sure you need the condition key checks?